rinex/src/qc/context.rs

```diff
@@ -1,61 +1,93 @@
 use horrorshow::{helper::doctype, RenderBox};
-use std::path::{Path, PathBuf};
+use std::path::PathBuf;
 
 use crate::prelude::GroundPosition;
 use crate::quality::HtmlReport;
-use crate::{Error, Rinex};
+use crate::Rinex;
 
 use sp3::prelude::SP3;
 
 #[derive(Default, Debug, Clone)]
-pub struct QcInputData {
-    /// File source path
+pub struct QcPrimaryData {
+    /// Source path
     pub path: PathBuf,
-    /// File Data
-    pub rinex: Rinex,
+    /// Data
+    pub data: Rinex,
 }
 
-impl QcInputData {
-    pub fn new(path: &str) -> Result<Self, Error> {
-        Ok(Self {
-            path: Path::new(path).to_path_buf(),
-            rinex: Rinex::from_file(path)?,
-        })
+#[derive(Default, Debug, Clone)]
+pub struct QcExtraData<T> {
+    /// Source paths
+    pub paths: Vec<PathBuf>,
+    /// Data
+    pub data: T,
+}
+
+impl<T> QcExtraData<T> {
+    /// Returns reference to Inner Data
+    pub fn data(&self) -> &T {
+        &self.data
+    }
+    /// Returns mutable reference to Inner Data
+    pub fn data_mut(&mut self) -> &mut T {
+        &mut self.data
+    }
+    /// Returns list of files that created this context
+    pub fn paths(&self) -> Vec<PathBuf> {
+        self.paths.clone()
```

Comment:
Hello @lnicola,

any recommendations to avoid this `.clone()` ? due to the fact we're retuning the List of PathBufs.

Not that this matters so much, this is only used in two locations
- when building the actual Context provided by the entire command line.
We need this to append a new path for example if this is the second file of this kind
- when reporting the final anlysis, in the report header we notify the file locations, and this is invoked once

Reply:
Easiest is to return a `&[PathBuf]`.

Reply:
Applied, it works. Not used to that form but it's simple